machine/v1beta1/types_gcpprovider.go

```diff
@@ -54,9 +54,20 @@ type GCPMachineProviderSpec struct {
 	// ProjectID is the project in which the GCP machine provider will create the VM.
 	// +optional
 	ProjectID string `json:"projectID,omitempty"`
+	// GuestAccelerators is a list of GPUs to be attached to the VM.
+	// +optional
+	GuestAccelerators []*GCPAcceleratorConfig `json:"guestAccelerators,omitempty"`
 	// Preemptible indicates if created instance is preemptible
 	// +optional
 	Preemptible bool `json:"preemptible,omitempty"`
+	// OnHostMaintenance Determines the behavior when a maintenance event occurs that might cause the instance to reboot (default MIGRATE).
+	// This is required to be set to TERMINATE if you want to provision machine with attached GPUs.
+	// +optional
+	OnHostMaintenance string `json:"onHostMaintenance,omitempty"`
+	// AutomaticRestart Determines the behavior when an instance crashes or is stopped by the system (default true).
+	// Can not be true with preepmtible instances.
+	// +optional
+	AutomaticRestart *bool `json:"automaticRestart,omitempty"`
```

Comment:
It's usually better to avoid `bool` in API definitions.  I see that other types the machine API uses it in many places though, so maybe it's all right here?

Reply:
follows https://pkg.go.dev/google.golang.org/api/compute/v1 convention

Reply:
> follows https://pkg.go.dev/google.golang.org/api/compute/v1 convention

We expect our CRDs to follow kubernetes API conventions.  While its not unheard of to have `*bool` as a type, I'm also interested in how that power is intended to be wielded by a user and how it is better than some kind of restart policy. 

CrashAction, with possible values of Restart, RemainStopped, "" appears more kube-like and explicitly describes the alternatives available.

Reply:
@SamuelStuchly Have you looked at whether the cluster api equivalent has this option? Curious to see how they handle it

Bool pointer would normally be used to allow three states, but I don't think we need that here, we could invert the type to make it just a bool eg `DisableAutomaticRestart bool // default false`, but I don't have an issue with moving this to the enum style suggested above either